### PR TITLE
VVVVVV: Update to 2.3

### PR DESCRIPTION
**Pull request by win8linux**
This updates the VVVVVV package to the 2.3 version as of 12 July 2020. There are too many changes from upstream to get through here, even when it comes to picking out highlights. In lieu of that, there will be some discussion regarding VSync and Haiku.

The game recently introduced VSync support, which broke it on Haiku due to the way it was implemented. The implementation was fixed so that the game would run in Haiku. However, enabling VSync would cause the game to crash until its settings were manually wiped; this persisted between reinstalls until deleted. To work around this, toggling VSync through the in-game settings was disabled. More detaiils can be found in [this issue](https://github.com/TerryCavanagh/VVVVVV/issues/345#issuecomment-652517276).

### CODE REVIEW

File: games-arcade/vvvvvv/vvvvvv-2.3~git.recipe [hunk 1]
```diff
@@ -10,10 +10,10 @@ purchasing a copy at https://thelettervsixtim.es."
 HOMEPAGE="https://thelettervsixtim.es"
 COPYRIGHT="Terry Cavanagh 2010-2020"
 LICENSE="VVVVVV Source Code License v1.0"
-REVISION="5"
-srcGitRev="5fdbaa007608e11d91a81dda7f0523255572018e"
+REVISION="6"
```
Comment (Begasus) [inline games-arcade/vvvvvv/vvvvvv-2.3~git.recipe:13]: Revision should be "1" here as this is a version bump

Comment (win8linux) [reply games-arcade/vvvvvv/vvvvvv-2.3~git.recipe:13]: The package has been updated multiple times in the past, mostly due to keeping up with changes upstream. 2.3 is the first known instance of a formal version bump, yet new fixes and features came well before it.

Comment (Begasus) [reply games-arcade/vvvvvv/vvvvvv-2.3~git.recipe:13]: Last official release is version 2.2, as this bumps to a git checkout to a "newer" version REVISION should be reset to mark a difference between last official version and a new release in progress

Comment (win8linux) [reply games-arcade/vvvvvv/vvvvvv-2.3~git.recipe:13]: Ah, ok then. My prior understanding of REVISION was a way for the package manager to determine whether a package is newer than the locally installed one.

Comment (Begasus) [reply games-arcade/vvvvvv/vvvvvv-2.3~git.recipe:13]: If no 2.2 version was released then yes, but there is an official tarbal for it, development went on after 2.2 preparing it for version 2.3, so a revision change would be needed

Comment (Begasus) [reply games-arcade/vvvvvv/vvvvvv-2.3~git.recipe:13]: Hold on :)

Comment (Begasus) [reply games-arcade/vvvvvv/vvvvvv-2.3~git.recipe:13]: Did you check if this change was imported in the official release (the commit is older than the official release)

Comment (win8linux) [reply games-arcade/vvvvvv/vvvvvv-2.3~git.recipe:13]: This was from 2 days ago while 2.3 was in development. Should I update it to the current commit presently?

Comment (Begasus) [reply games-arcade/vvvvvv/vvvvvv-2.3~git.recipe:13]: Looking at their github page they announced a 2.2 release 18hours ago? 
https://github.com/TerryCavanagh/VVVVVV/releases/tag/2.2

Comment (Begasus) [reply games-arcade/vvvvvv/vvvvvv-2.3~git.recipe:13]: If so then yes, bump to the latest commit there to keep version 2.3

Comment (win8linux) [reply games-arcade/vvvvvv/vvvvvv-2.3~git.recipe:13]: The commit which this is based on is on 2.3 work. One of the maintainers mentions this in the issue linked in the PR description:
> I'd be up for disabling the option in Haiku builds for 2.3

Do I still need to update the recipe to the lates upstream commit?

Comment (Begasus) [reply games-arcade/vvvvvv/vvvvvv-2.3~git.recipe:13]: Ok, if this is working for you then keep the current upstream commit and version 2.3, only revision should still be changed then

**Comment by Begasus** [comment]
Has this been checked on 32bit and 64bit?

**Comment by win8linux** [comment]
> Has this been checked on 32bit and 64bit?

Only 64-bit at the moment. This is why `!x86_gcc2` is marked as such in the recipe.

**Comment by Begasus** [comment]
Hence why we use secondary architecture :)
On that note, crashes on 32bit ...
```
./VVVVVV 
Base directory: /boot/home/config/settings/VVVVVV/
Save directory: /boot/home/config/settings/VVVVVV/saves/
Level directory: /boot/home/config/settings/VVVVVV/levels/
gamecontrollerdb.txt not found!
                
                
                       VVVVVV
                
                
                  8888888888888888  
                88888888888888888888
                888888    8888    88
                888888    8888    88
                88888888888888888888
                88888888888888888888
                888888            88
                88888888        8888
                  8888888888888888  
                      88888888      
                  8888888888888888  
                88888888888888888888
                88888888888888888888
                88888888888888888888
                8888  88888888  8888
                8888  88888888  8888
                    888888888888    
                    8888    8888    
                  888888    888888  
                  888888    888888  
                  888888    888888  
                
                
Quick Save Not Found
Teleporter Save Not Found
Unable to open file mmmmmm.vvv
The complete reloaded file size: 61651755
data/music/0levelcomplete.ogg unpacked
data/music/1pushingonwards.ogg unpacked
data/music/2positiveforce.ogg unpacked
data/music/3potentialforanything.ogg unpacked
data/music/4passionforexploring.ogg unpacked
data/music/5intermission.ogg unpacked
data/music/6presentingvvvvvv.ogg unpacked
data/music/7gamecomplete.ogg unpacked
data/music/8predestinedfate.ogg unpacked
data/music/9positiveforcereversed.ogg unpacked
data/music/10popularpotpourri.ogg unpacked
data/music/11pipedream.ogg unpacked
data/music/12pressurecooker.ogg unpacked
data/music/13pacedenergy.ogg unpacked
data/music/14piercingthesky.ogg unpacked
data/music/predestinedfatefinallevel.ogg unpacked
Kill Thread
/boot/system/apps/vvvvvv>
```

**Comment by win8linux** [comment]
Should I completely remove `x86_gcc2` from the recipe in the meantime?

**Comment by Begasus** [comment]
> Should I completely remove `x86_gcc2` from the recipe in the meantime?

No, x86_gcc2 is the primary architecture (gcc2), it can't be build for it hence the "!x86_gcc2", it can however be build with gcc8 for "secondary architecture" (hence "x86" there)

**Comment by win8linux** [comment]
Well if it doesn't work on 32-bit ATM, then maybe the recipe should be changed to be 64-bit only for the meantime?

**Comment by win8linux** [comment]
Made the decision to make the recipe build only for 64-bit in the meantime.

**Comment by Begasus** [comment]
**2.0~git** still works on 32bit, so this should be kept, if you want to disable the build for **2.3~git** (on 32bit) then you need to change:
```
ARCHITECTURES="!x86_gcc2 x86_64"
SECONDARY_ARCHITECTURES="!x86"
```

**Comment by win8linux** [comment]
If the 2.0~git is kept, should it be restricted to 32-bit only to prevent conflicts with 2.3-git on 64-bit systems?

**Comment by Begasus** [comment]
You still need to leave the **2.0~git** version around, for 64bit you need not worry, it will build the latest one at the buildbots so only one will be available (latest working version)

**Pull request merged**